messageapi/src/test/java/fi/hel/verkkokauppa/message/service/MessageServiceTest.java

```diff
@@ -0,0 +1,117 @@
+package fi.hel.verkkokauppa.message.service;
+
+import com.dumbster.smtp.SimpleSmtpServer;
+import com.dumbster.smtp.SmtpMessage;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fi.hel.verkkokauppa.message.dto.MessageDto;
+import fi.hel.verkkokauppa.message.model.Message;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import javax.mail.Session;
+import javax.mail.internet.MimeMessage;
+import java.net.URI;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class MessageServiceTest {
+    /**
+     * Elasticsearch version which should be used for the Tests
+     */
+    private static final String ELASTICSEARCH_VERSION = "7.9.2";
+    private static final DockerImageName ELASTICSEARCH_IMAGE =
+            DockerImageName
+                    .parse("docker.elastic.co/elasticsearch/elasticsearch")
+                    .withTag(ELASTICSEARCH_VERSION);
+
+    // bind the above RANDOM_PORT
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        // Create the elasticsearch container.
+        try (ElasticsearchContainer container = new ElasticsearchContainer(ELASTICSEARCH_IMAGE)) {
```

Comment:
We should keep unit tests simple (logic tests and mocks only) and run integration tests separately (too heavy to run on every build).

Reply:
Good point, 

- [ ]  going to remove this dependency because it is not needed yet.